pkg/cmd/pulumi/config.go

```diff
@@ -240,33 +260,37 @@ func copyEntireConfigMap(currentStack backend.Stack,
 
 	encrypter, _, cerr := getStackEncrypter(destinationStack, destinationProjectStack)
 	if cerr != nil {
-		return cerr
+		return false, cerr
 	}
 
 	newProjectConfig, err := currentConfig.Copy(decrypter, encrypter)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	var requiresSaving bool
 	for key, val := range newProjectConfig {
 		err = destinationProjectStack.Config.Set(key, val, false)
 		if err != nil {
-			return err
+			return false, err
 		}
 		requiresSaving = true
 	}
 
-	// The use of `requiresSaving` here ensures that there was actually some config
-	// that needed saved, otherwise it's an unnecessary save call
-	if requiresSaving {
-		err := saveProjectStack(destinationStack, destinationProjectStack)
-		if err != nil {
-			return err
+	if currentEnvironments != nil && len(currentEnvironments.Imports()) > 0 {
+		if destinationProjectStack.Environment != nil {
+			imports := currentEnvironments.Imports()
+			for _, env := range imports {
+				destinationProjectStack.Environment.Append(env)
+			}
+		} else {
+			destinationProjectStack.Environment = currentEnvironments
 		}
+
+		requiresSaving = true
 	}
 
-	return nil
+	return requiresSaving, nil
```

Comment:
Note that this change had a cascading effect on another place. Though I am not a huge fan of this approach I feel like this is simple and makes this function testable through unit testing.

I am also thinking if I should rename this method from `copyEntireConfigMap` to something like `getMergdConfigMap`?

Reply:
@justinvp @pgavlin before merging this though, I want to draw attention to this comment in case you missed it. Would love to know your thoughts.

Reply:
Returning `requiresSaving` and making it so callers have to call save doesn't seem the best, but I don't feel strongly about it and agree it's nice that it makes it easier to unit test. I'm fine with it.

I also don't feel strongly about the function name.